modules/health_quest/app/services/health_quest/patient_generated_data/options_builder.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+module HealthQuest
+  module PatientGeneratedData
+    class OptionsBuilder
+      attr_reader :user, :filters
+
+      def self.manufacture(user, filters)
+        new(user, filters)
+      end
+
+      def initialize(user, filters)
+        @user = user
+        @filters = filters
+      end
+
+      def to_hash
+        if appointment_id.present?
+          { subject: appointment_reference }
+        else
+          { author: user.icn }
+        end
+      end
+
+      def appointment_reference
+        "#{Settings.hqva_mobile.url}/appointments/v1/patients/#{user.icn}/Appointment/#{appointment_id}"
+      end
```

Comment:
What does "reference" mean in this context? It seems like we're building a URL for a given appointment, could this be called `appointment_url` ? Or `appointment_path` ?

Reply:
Hi, good question. The method name refers to a "reference" in a FHIR resource in the PGD; The QuestionnaireResponse resource' `subject reference` to be specific. I thought naming it `appointment_reference` would capture the intent but now that you mention it I can see that naming it as is can be a bit confusing or misleading. Since these are not appointments or appointment related resources I don't think calling them `appointment_<suffix>' makes sense in this context. How about `subject_reference` ? I could see the need for different types of subject_references for different types of resources in the future.

Reply:
Ok - so it's a FHIR-specific term. The name change and docs you made make it much clearer, thank you!